<!-- LAUNCH FILE FOR MISSION CONTROL -->

<launch>

  <!-- joy node -->
  <node respawn="true"  pkg="joy"
        type="joy_node" name="IRIS_joy" >
    <param name="dev" value="/dev/input/js0" type="string" />
  </node>

  <!-- joy message cleanup node -->
  <node respawn="true" pkg="manual_robot_control" type="joy_filter"
        name="IRIS_joy_filter" output="screen">
    <param name="threshold/x"       value="0.10" />
    <param name="threshold/z"       value="0.12" />
    <param name="threshold/jitter"  value="0.05" />
    <param name="max_rate"          value="10"  type="double" />
    <param name="min_rate"          value="2"   type="double" />
    <param name="topic"             value="/IRIS/joy_filtered"/>
  </node>

  <!-- heartbeat for FSM -->
  <node respawn="true" pkg="finite_state_machine" type="trigger"
        name="FSM_heartbeat" >
    <param name="frequency" value="2" />
    <param name="topic"     value="/IRIS/heartbeat" />
  </node>

</launch>
